eservice/lib/libpdo_enclave/CMakeLists.txt

```diff
@@ -125,9 +125,22 @@ SET(CMAKE_SHARED_LIBRARY_PREFIX "")
 
 ADD_LIBRARY(${PROJECT_NAME} SHARED  ${PROJECT_HEADERS} ${PROJECT_SOURCES} ${PROJECT_EDGE_SOURCES} ${PROJECT_EDL})
 SGX_SIGN_ENCLAVE(${PROJECT_NAME} ${PDO_ENCLAVE_CODE_SIGN_PEM} ${PROJECT_CONFIG})
+
+TARGET_LINK_LIBRARIES(${PROJECT_NAME} "${PDO_TOP_DIR}/common/build/libtpdo-common.a")
+TARGET_LINK_LIBRARIES(${PROJECT_NAME} "${PDO_TOP_DIR}/common/build/libpdo_contracts.a")
+IF ($ENV{PDO_INTERPRETER} STREQUAL "gipsy")
```

Comment:
Future enhancement:
since the interpreter is beautifully abstracted now... it does not make much sense to keep track of different libraries, which expose the same interface. What I mean is that: from `common/interpreter` we should produce a `libcontractinterpreter.a` artifact (which combines either gipsy/tinyscheme or w/iwasm) and hide the interpreter-specific libraries from the eservice.

Reply:
Slightly different approach: produce a single contract interpreter library per interpreter type. That is, as the developer, I'd like to be able to see whether I'm importing the PDO interpreter for wawaka or gipsy since I presumably care about whether I will be running scheme or WASM contracts. We can then still hide the interpreter-specific dependencies (e.g. `libtinyscheme.a` vs `libiwasm.a`), as you suggest.

Reply:
good suggestion though i have a feeling it will be easier to describe than implement. could you add an issue?

Reply:
Yeah... you're probably right. I'll add an issue for this.